python/tubix20_reframer.py

```diff
@@ -0,0 +1,176 @@
+# Copyright 2025 Fabian P. Schmidt <kerel@mailbox.org>
+#
+# This file is part of gr-satellites
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+"""
+References:
+[1]: TechnoSat_TUBIN_Telemetry_Format
+     <https://www.static.tu.berlin/fileadmin/www/10002275/
+     Amateur_Radio/TechnoSat_Telemetry_Format.ods>
+[2]: <https://destevez.net/2016/09/some-notes-on-beesat-and-mobitex-nx/>
+"""
+from gnuradio import gr
+import pmt
+
+from .check_eseo_crc import crc16_ccitt_zero as crc16_ccitt_zero
+
+
+class tubix20_reframer(gr.basic_block):
+    """
+    Input:
+        PDUs with decoded mobitex data blocks.
+        Each PDU must have a 'block id' tag.
+        The first PDU must have the following additional tags:
+            - 'num_blocks' tag: Number of blocks for this frame
+            - 'frame_header' tag: Header of this frame (typically 11 bytes)
+                - 2 control bytes
+                - 1 control bytes FEC
+                - 6 callsign bytes
+                - 2 callsign CRC bytes
+    Output:
+        PDUs with Tubix20 "Master Frames"
+        - master frame header (typically 10 bytes)
+            - 2 control bytes
+            - 6 callsign bytes
+            - 2 callsign CRC bytes
+        - master frame body
+            - N x 18 bytes
+        - master frame appendix
+            - 1 byte "ground station quality byte"
+            - 4 bytes "error marker"
+            - 1 byte "ground station TNC temperature"
```

Comment:
Maybe add that the quality byte and TNC temperature are unused by this implementation and filled with placeholder values.